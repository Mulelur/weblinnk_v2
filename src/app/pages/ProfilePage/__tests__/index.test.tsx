import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { themes } from 'styles/theme/themes';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';

import { ProfilePage } from '..';

const renderPage = () =>
  renderer.create(
    <MemoryRouter>
      <ThemeProvider theme={themes.light}>
        <HelmetProvider>
          <ProfilePage />
        </HelmetProvider>
      </ThemeProvider>
    </MemoryRouter>,
  );

describe('<ProfilePage />', () => {
  it('should match snapshot', () => {
    const notFoundPage = renderPage();
    expect(notFoundPage.toJSON()).toMatchSnapshot();
  });
});
