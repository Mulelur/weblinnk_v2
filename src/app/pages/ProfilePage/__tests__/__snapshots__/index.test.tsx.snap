// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> should match snapshot 1`] = `
Array [
  .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  border-radius: 0.25rem;
  border: 1px solid rgba(58,52,51,0.05);
  background-color: rgba(255,255,255,1);
  background-color: transparent;
  color: rgba(58,52,51,1);
  border: none;
}

.c4:focus {
  outline: none;
  border: 1px solid rgba(215,113,88,1);
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
  color: rgba(215,113,88,1);
}

.c6 {
  color: rgba(215,113,88,1);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.15rem 0.3rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6:hover {
  opacity: 0.8;
}

.c6:active {
  opacity: 0.4;
}

.c6 .icon {
  margin-right: 0.2rem;
}

.c1 {
  display: none;
}

.c2 {
  color: rgba(215,113,88,1);
  padding-right: 0.8rem;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
  width: 100%;
  padding-left: 10rem;
  padding-right: 10rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {
  .c1 {
    display: block;
  }
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    padding-left: 1rem;
    padding-right: 3rem;
  }
}

@media only screen and (min-width:600px) {
  .c0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding-left: 17rem;
    padding-right: 7rem;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    padding-left: 18rem;
    padding-right: 8rem;
  }
}

@media only screen and (max-width:600px) {

}

@media only screen and (min-width:600px) {

}

<div
    className="c0"
    onClick={[Function]}
  >
    <div
      className="c1"
    >
      <div
        className=""
      >
        <div
          className="c2"
        >
          <svg>
            menu-icon.svg
          </svg>
        </div>
      </div>
    </div>
    <div
      className="c3"
    >
      <input
        className="c4"
        placeholder="search"
      />
    </div>
    <div
      className="c5"
    >
      <a
        className="c6"
        href="/profile"
        rel="noopener noreferrer"
        target="_blank"
        title="Documentation Page"
      >
        <svg>
          account-icon.svg
        </svg>
      </a>
      <a
        className="c6"
        href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
        rel="noopener noreferrer"
        target="_blank"
        title="Github Page"
      >
        <svg>
          bell-icon.svg
        </svg>
      </a>
    </div>
  </div>,
  .c1 {
  color: rgba(58,52,51,1);
  font-size: 1.25rem;
}

.c2 {
  font-size: 0.8rem;
  line-height: 1.5;
  color: rgba(58,52,51,0.7);
  margin: 0.225rem 0 1.5rem 0;
}

.c4 {
  border-radius: 8px;
  border-color: rgba(58,52,51,0.05);
  background-color: rgba(255,255,255,1);
  box-shadow: rgba(215,113,88,1) 0px 0px 0px 1px inset,rgb(38 38 44 / 7%) 0px 1px 3px;
  padding: 0.5rem 1.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: rgba(58,52,51,1);
  font-weight: 500;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  margin: 0.5rem;
  padding: 1.3rem;
  width: 100%;
}

.c14 {
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: rgba(215,113,88,1);
  font-size: 1.05rem;
  padding: 0.5rem;
  margin: 0.5rem 0;
  border: 1px solid rgba(58,52,51,0.12);
  background-color: rgba(255,255,255,1);
  border-radius: 0.45rem;
  width: 100%;
  height: 2.6rem;
}

.c14:focus {
  outline: none;
  color: rgba(215,113,88,1);
  border: 1px solid rgba(215,113,88,1);
}

.c13 {
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: rgba(58,52,51,0.7);
  font-size: 0.95rem;
  font-weight: 700;
}

.c11 {
  width: 100%;
  padding-bottom: 2rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c15 {
  margin: 1rem;
}

.c16 {
  background-color: rgba(255,255,255,1);
  color: rgba(215,113,88,1);
  display: table;
  border: none;
  position: relative;
  font-weight: 600;
  font-size: 14px;
  font-family: inherit;
  margin: 0px;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  text-align: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 4px;
  line-height: 20px;
  -webkit-letter-spacing: -0.125px;
  -moz-letter-spacing: -0.125px;
  -ms-letter-spacing: -0.125px;
  letter-spacing: -0.125px;
  padding: 10px var(--button-padding-h,24px);
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease 0s,box-shadow 0.2s ease 0s, color 0.2s ease 0s,-webkit-transform 0.2s ease 0s;
  -webkit-transition: background-color 0.2s ease 0s,box-shadow 0.2s ease 0s, color 0.2s ease 0s,transform 0.2s ease 0s;
  transition: background-color 0.2s ease 0s,box-shadow 0.2s ease 0s, color 0.2s ease 0s,transform 0.2s ease 0s;
  -webkit-transform: scale(var(--button-scale,1)) translateZ(0);
  -ms-transform: scale(var(--button-scale,1)) translateZ(0);
  transform: scale(var(--button-scale,1)) translateZ(0);
  box-shadow: 0 0 0 var(--button-outline,0px) var(--button-outline-color,#5c89de4d), inset 0 -1px 1px 0 var(--button-inner-shadow,rgba(38 38 44 / 15%)), inset 0 var(--button-shine-y,1px) 0 0 var(--button-shine,rgba(255 255 255 / 25%)), 0 1px 2px 0 var(--button-shadow,rgba(38 38 44 / 25%));
}

.c3 {
  margin: 1rem auto;
  padding: 0 1.5rem;
  box-sizing: content-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  height: 3.2rem;
  width: 3.2rem;
  color: rgba(215,113,88,1);
  background-color: rgba(255,255,255,1);
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.6rem;
  box-shadow: 0px 1px 3px 0px rgba(0,0,0,0.1),0px 1px 2px 0px rgba(0,0,0,0.06);
}

.c5 {
  width: 100%;
}

@media only screen and (max-width:768px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (max-width:768px) {

}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {

}

@media only screen and (max-width:600px) {

}

@media only screen and (min-width:600px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:992px) {

}

@media only screen and (max-width:600px) {
  .c3 {
    padding: 0 0.9rem;
  }
}

@media only screen and (min-width:600px) {
  .c3 {
    padding: 0 1.1rem;
  }
}

<div
    className="c0"
  >
    <h1
      className="c1"
    >
      Get The Right Plan For You
    </h1>
    <p
      className="c2"
    >
      All plans including 14-days free trial from us
    </p>
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              A
            </div>
            <h4
              className="c8"
            >
              Profile
            </h4>
          </div>
          <div
            className="c9"
          >
            <div
              className="c10"
            >
              <form
                className="c11"
              >
                <div
                  className="c9"
                >
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 1
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                  <div
                    className="c15"
                  />
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 2
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="c9"
                >
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 1
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                  <div
                    className="c15"
                  />
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 2
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c15"
      />
      <h1
        className="c1"
      >
        Password
      </h1>
      <p
        className="c2"
      >
        You can change your Password here!
      </p>
      <div
        className="c4"
      >
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <div
              className="c7"
            >
              <svg>
                secure-icon.svg
              </svg>
            </div>
            <h4
              className="c8"
            >
              Password
            </h4>
          </div>
          <div
            className="c9"
          >
            <div
              className="c10"
            >
              <form
                className="c11"
              >
                <div
                  className="c9"
                >
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 1
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                  <div
                    className="c15"
                  />
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 2
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="c9"
                >
                  <div
                    className="c12"
                  >
                    <label
                      className="c13"
                    >
                      Link 1
                    </label>
                    <input
                      className="c14"
                      placeholder="Enter something here!"
                      type="text"
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c10"
      >
        <button
          className="c16"
        >
          Save
        </button>
      </div>
    </div>
  </div>,
]
`;
